rust/cubesqlplanner/cubesqlplanner/src/planner/base_measure.rs

```diff
@@ -138,36 +136,28 @@ impl BaseMeasure {
             name.clone(),
             expression,
             member_expression_definition.clone(),
+            query_tools.base_tools().clone(),
         )?;
         let full_name = member_expression_symbol.full_name();
-        let member_evaluator = Rc::new(MemberSymbol::MemberExpression(member_expression_symbol));
+        let member_evaluator = MemberSymbol::new_member_expression(member_expression_symbol);
         let default_alias = PlanSqlTemplates::alias_name(&name);
         Ok(Rc::new(Self {
             measure: full_name,
             query_tools,
             member_evaluator,
-            definition: None,
             cube_name,
             name,
             member_expression_definition,
             default_alias,
         }))
     }
 
-    pub fn can_used_as_addictive_in_multplied(&self) -> Result<bool, CubeError> {
-        let measure_type = self.measure_type();
-        let res = if measure_type == "countDistinct" || measure_type == "countDistinctApprox" {
-            true
-        } else if measure_type == "count" {
-            if let Some(definition) = &self.definition {
-                !definition.has_sql()?
-            } else {
-                false
-            }
+    pub fn can_used_as_addictive_in_multplied(&self) -> bool {
+        if let Ok(measure_symbol) = self.member_evaluator.as_measure() {
+            measure_symbol.can_used_as_addictive_in_multplied()
```

Comment:
Would be nice to rename this too:
```suggestion
            measure_symbol.can_be_used_as_additive_in_multplied()
```